app/controllers/api/v01/bibs_controller.rb

```diff
@@ -8,23 +8,50 @@ class Api::V01::BibsController < ApplicationController
   def create_or_update_teacher_sets
     error_code_and_message = validate_request
     if error_code_and_message.any?
+      AdminMailer.failed_bibs_controller_api_request(@request_body, error_code_and_message, action_name)
       render_error(error_code_and_message)
-      return
     end
+    return if error_code_and_message.any?
```

Comment:
Line 11 and 14 have the same conditional. I'd suggest adding the return in the first if statement after line 12.